From bdeef55452e7e7ff8338c44196b053a3d55c0c13 Mon Sep 17 00:00:00 2001
From: Fancy Fang <chen.fang@nxp.com>
Date: Tue, 17 Apr 2018 10:56:07 +0800
Subject: [PATCH 3607/5242] MLK-18045-1 drm/imx: dcss: define 'struct
 dma_metadata' for dec400d config

commit  e920ac74adff4eb819d83be0f5a03d7598d0da37 from
https://source.codeaurora.org/external/imx/linux-imx.git

Define a new struct 'dma_metadata' to hold the config parameters
for DEC400D. This struct data should be passed in from the fb's
first gem_obj's 'dma_buf' field.

Signed-off-by: Fancy Fang <chen.fang@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 include/video/imx-dcss.h |   15 +++++++++++++++
 1 file changed, 15 insertions(+)

diff --git a/include/video/imx-dcss.h b/include/video/imx-dcss.h
index 253dff8..2b63cbc 100644
--- a/include/video/imx-dcss.h
+++ b/include/video/imx-dcss.h
@@ -25,6 +25,21 @@ struct dcss_client_platformdata {
 	struct device_node *of_node;
 };
 
+#define VIV_VIDMEM_METADATA_MAGIC fourcc_code('v', 'i', 'v', 'm')
+
+struct dma_metadata {
+	uint32_t magic;
+	int32_t  ts_buf_fd;
+	void *ts_dma_buf;
+
+	uint32_t fc_enabled;
+	uint32_t fc_value;
+	uint32_t fc_value_upper;
+
+	uint32_t compressed;
+	uint32_t compressed_format;
+};
+
 /* COMMON */
 int dcss_vblank_irq_get(struct dcss_soc *dcss);
 void dcss_vblank_irq_enable(struct dcss_soc *dcss, bool en);
-- 
1.7.9.5

